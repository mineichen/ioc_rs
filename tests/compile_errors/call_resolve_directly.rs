use ioc_rs::{ServiceCollection, Dynamic, resolvable::Resolvable};

fn main() {
    let container = ServiceCollection::new();
    let provider = container.build().expect("Expected to have all dependencies");
    Dynamic::<std::sync::Arc<i32>>::resolve(&provider);
}