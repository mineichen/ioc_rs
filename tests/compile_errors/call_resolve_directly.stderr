error[E0603]: module `resolvable` is private
  --> $DIR/call_resolve_directly.rs:1:42
   |
1  | use ioc_rs::{ServiceCollection, Dynamic, resolvable::Resolvable};
   |                                          ^^^^^^^^^^ private module
   |
note: the module `resolvable` is defined here
  --> $DIR/lib.rs:47:1
   |
47 | mod resolvable;
   | ^^^^^^^^^^^^^^^

error[E0599]: no function or associated item named `resolve` found for struct `Dynamic<Arc<i32>>` in the current scope
 --> $DIR/call_resolve_directly.rs:6:37
  |
6 |     Dynamic::<std::sync::Arc<i32>>::resolve(&provider);
  |                                     ^^^^^^^ function or associated item not found in `Dynamic<Arc<i32>>`
  |
  = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
  |
1 | use ioc_rs::resolvable::Resolvable;
  |
